dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -215,6 +239,70 @@ internal class DatadogRumMonitor(
 
     // region Internal
 
+    private fun updateAttributesWithActionId(
+        attributes: Map<String, Any?>
+    ): Map<String, Any?> {
+        val actionId = getActiveActionId()
+        return if (actionId == null) {
+            attributes
+        } else {
+            attributes.toMutableMap()
+                .apply {
+                    put(RumEventSerializer.TAG_EVENT_USER_ACTION_ID, actionId.toString())
+                }
+        }
+    }
+
+    private fun getActiveActionId(): UUID? {
+        val nanoTime = System.nanoTime()
+        val lastAction = lastActionRelatedEventNs.get()
+        val actionEvent = activeActionEvent
+        val actionData = activeActionData
+        val isLastEventRecent = (nanoTime - lastAction) < ACTION_INACTIVITY_NS
+        ongoingUserActionResources.removeAll { it.get() == null }
+        val hasUnclosedResources = ongoingUserActionResources.isNotEmpty()
+        val isWithinActionScope = isLastEventRecent || hasUnclosedResources
+
+        return if (
+            actionEvent != null &&
+            actionData != null &&
+            isWithinActionScope
+        ) {
+            lastActionRelatedEventNs.set(nanoTime)
+            actionData.id
+        } else {
+            sendUserAction(actionEvent, actionData, lastAction)
+            activeActionEvent = null
+            activeActionData = null
+            null
```

Comment:
maybe just call the `sendUnclosedAction` for better visibility ?

Reply:
Hmmm… Not a fan on doing this, it will do a get on all fields again for nothing.